src/front/spv/flow.rs

```diff
@@ -217,33 +217,35 @@ impl FlowGraph {
     }
 
     /// Traverses the flow graph and returns a list of Naga's statements.
-    pub(super) fn to_naga(&self) -> Result<crate::Block, Error> {
+    pub(super) fn convert_to_naga(&mut self) -> Result<crate::Block, Error> {
         self.naga_traverse(node_index(0), None)
     }
 
     fn naga_traverse(
-        &self,
+        &mut self,
         node_index: BlockNodeIndex,
         stop_node_index: Option<BlockNodeIndex>,
     ) -> Result<crate::Block, Error> {
         if stop_node_index == Some(node_index) {
             return Ok(vec![]);
         }
 
-        let node = &self.flow[node_index];
+        let terminator = self.flow[node_index].terminator.clone();
```

Comment:
we need something like `let (node_block, terminator) = self.flow[node_index].consume();`, so that we can have some condition inside it to check if the node was already consumed previously